vortex-expr/src/binary.rs

```diff
@@ -82,6 +83,66 @@ pub(crate) mod proto {
     }
 }
 
+impl AnalysisExpr for BinaryExpr {
+    fn prune_expr(&self, catalog: &mut dyn StatsCatalog) -> Option<ExprRef> {
+        match self.operator {
+            Operator::Eq => {
+                let min_lhs = self.lhs.min(catalog);
+                let max_lhs = self.lhs.max(catalog);
+
+                let min_rhs = self.rhs.min(catalog);
+                let max_rhs = self.rhs.max(catalog);
+
+                let left = min_lhs.zip_with(max_rhs, gt);
+                let right = min_rhs.zip_with(max_lhs, gt);
+                left.into_iter().chain(right).reduce(or)
+            }
+            Operator::NotEq => {
+                let min_lhs = self.lhs.min(catalog);
+                let max_lhs = self.lhs.max(catalog);
+
+                let min_rhs = self.rhs.min(catalog);
+                let max_rhs = self.rhs.max(catalog);
+
+                let left = min_lhs.zip_with(max_rhs, eq);
+                let right = max_lhs.zip_with(min_rhs, eq);
+                left.into_iter().chain(right).reduce(and)
+            }
+            Operator::Gt | Operator::Gte => self.lhs.max(catalog).and_then(|max| {
+                self.rhs.min(catalog).map(|min| {
+                    let inverse = self
+                        .operator
+                        .inverse()
+                        .vortex_expect("inverse of gt & gt_eq defined");
+                    BinaryExpr::new_expr(max, inverse, min)
+                })
+            }),
+            Operator::Lt | Operator::Lte => self.lhs.min(catalog).and_then(|min| {
+                self.rhs.max(catalog).map(|max| {
+                    let inverse = self
+                        .operator
+                        .inverse()
+                        .vortex_expect("inverse of le & lte defined");
+                    BinaryExpr::new_expr(min, inverse, max)
+                })
+            }),
```

Comment:
```suggestion
            Operator::Lt  => gte(self.lhs.min(catalog)?, self.rhs.max(catalog)?),
            Operator::Lte => gt(self.lhs.min(catalog)?, self.rhs.max(catalog)?),
```

I know this is mostly a copy from the old code but the noise of the inverse and vortex_expect hurts legibility, imo pretty badly.